main/main.go

```diff
@@ -251,21 +261,28 @@ func runRelayer(logger logging.Logger,
 				)
 				continue
 			}
-		case err := <-subscriber.Err():
+		case err := <-relayer.Subscriber.Err():
 			logger.Error(
 				"Received error from subscribed node",
 				zap.String("originChainID", sourceSubnetInfo.BlockchainID),
 				zap.Error(err),
 			)
-			err = subscriber.Subscribe()
+			err = relayer.ReconnectToSubscriber()
 			if err != nil {
 				logger.Error(
```

Comment:
We should mark the relayer as unhealthy in this case too.

Reply:
The reconnect function marks the relayer as unhealthy/healthy

Reply:
I'll move the set to unhealthy outside of `ReconnectToSubscriber`